hudi-common/src/main/java/org/apache/hudi/common/model/BaseAvroPayload.java

```diff
@@ -39,13 +40,19 @@
    */
   protected final Comparable orderingVal;
 
+  /**
+   * Schema used to convert avro to bytes.
+   */
+  protected final Schema writerSchema;
```

Comment:
This would increase the memory footprint (and increase I/O in shuffle stages if we introduce schema at payload level. 
You may want to introduce another class (similar to BaseAvroPayload) but also tracking schema at record level cc @n3nash

Reply:
@bvaradar @n3nash Yes, i had this concern raised in one of my earlier comments and wasn't sure how to handle it.

Thanks for the suggestion but it is not clear to me yet on how to handle this. Could one of you provide little bit more information.

Reply:
You can introduce another base class BaseAvroPayloadWithSchema which extends from BaseAvroPayload and stores the schema. This will be the base class for any new implementation which needs to store schema as part of pyload

Reply:
Thanks. I have updated it now